src/BotwinExtensions.cs

```diff
@@ -176,19 +208,22 @@ public static void AddBotwin(this IServiceCollection services)
 
             foreach (var module in modules)
             {
+                logger?.LogTrace($"Found {module.FullName}");
```

Comment:
These are pretty noisy. Maybe collect the module names, and then only log once with a list of detected modules?

Also: Use structured logging instead of string interpolation.